File: src/main/java/com/my/list/type/text/Text.java
package com.my.list.type.text;

import com.fasterxml.jackson.annotation.JsonTypeName;
import com.my.list.entity.ExtraData;
import lombok.Data;
import lombok.ToString;

@Data
@ToString
@JsonTypeName(value = "text")
public class Text implements ExtraData {

    private Long id = null;
    private String content = null;

}

File: src/main/java/com/my/list/type/text/Text.kt
package com.my.list.type.text

import com.fasterxml.jackson.annotation.JsonTypeName
import com.my.list.domain.ExtraData

@JsonTypeName(value = "text")
class Text(
    var id: Long? = null,
    var content: String? = null
): ExtraData {
    override fun toString(): String {
        return "Text[$id,$content]"
    }
    override fun getExtraId(): Long? { return id; }
    override fun setExtraId(id: Long?) { this.id = id; }
    override fun toMap(): Map<String, Any?> {
        val map = HashMap<String, Any?>()
        map["text_id"] = id
        map["text_content"] = content
        return map
    }
}
